slatedb/src/db_snapshot.rs

```diff
@@ -0,0 +1,78 @@
+use bytes::Bytes;
+use std::ops::RangeBounds;
+use std::sync::Arc;
+
+use crate::config::{ReadOptions, ScanOptions};
+use crate::db_iter::DbIterator;
+use crate::error::SlateDBError;
+use crate::Db;
+
+pub struct DbSnapshot {}
+
+impl DbSnapshot {
+    pub(crate) fn new(db: Arc<Db>) -> Self {
+        todo!()
+    }
+
+    /// Get a value from the snapshot with default read options.
+    ///
+    /// ## Arguments
+    /// - `key`: the key to get
+    ///
+    /// ## Returns
+    /// - `Result<Option<Bytes>, SlateDBError>`: the value if it exists, None otherwise
+    pub async fn get<K: AsRef<[u8]> + Send>(&self, key: K) -> Result<Option<Bytes>, SlateDBError> {
```

Comment:
Would it be possible to introduce a trait for the read methods, that we can reuse for db/snapshot ..? eg DbReadable?